zfit_physics/models/pdf_novosibirsk.py

```diff
@@ -0,0 +1,169 @@
+from typing import Optional
+
+import numpy as np
+import tensorflow as tf
+import zfit
+import zfit.z.numpy as znp
+from zfit import z
+from zfit.core.space import ANY_LOWER, ANY_UPPER, Space
+from zfit.util import ztyping
+
+
+@z.function(wraps="tensor")
+def novosibirsk_pdf(x, peak, width, tail):
+    """Calculate the Novosibirsk PDF.
+
+    Args:
+        x: value(s) for which the PDF will be calculated.
+        peak: peak of the distribution.
+        width: width of the distribution.
+        tail: tail of the distribution.
+
+    Returns:
+        `tf.Tensor`: The calculated PDF values.
+
+    Notes:
+        Function taken from H. Ikeda et al. NIM A441 (2000), p. 401 (Belle Collaboration)
+        Based on code from `ROOT <https://root.cern.ch/doc/master/Novosibirsk_8cxx_source.html>`_
+    """
+    x = z.unstack_x(x)
+
+    cond1 = znp.less(znp.abs(tail), 1e-7)
+    arg = 1.0 - (x - peak) * tail / width
+
+    cond2 = znp.less(arg, 1e-7)
+    log_arg = znp.log(arg)
+    xi = 2 * np.sqrt(np.log(4.0))
+
+    width_zero = (2.0 / xi) * znp.arcsinh(tail * xi * 0.5)
+    width_zero2 = width_zero**2
+    exponent = (-0.5 / width_zero2 * log_arg**2) - (width_zero2 * 0.5)
+
+    gauss = znp.exp(-0.5 * ((x - peak) / width) ** 2)
+
+    return znp.where(cond1, gauss, znp.where(cond2, 0.0, znp.exp(exponent)))
+
+
+def novosibirsk_integral(limits: ztyping.SpaceType, params: dict, model) -> tf.Tensor:
+    """Calculates the analytic integral of the Novosibirsk PDF.
+
+    Args:
+        limits: An object with attribute limit1d.
+        params: A hashmap from which the parameters that defines the PDF will be extracted.
+        model: Will be ignored.
+
+    Returns:
+        The calculated integral.
+    """
+    lower, upper = limits.limit1d
+    peak = params["mu"]
+    width = params["sigma"]
+    tail = params["lambd"]
+
+    return novosibirsk_integral_func(peak=peak, width=width, tail=tail, lower=lower, upper=upper)
+
+
+@z.function(wraps="tensor")
+def novosibirsk_integral_func(peak, width, tail, lower, upper):
+    """Calculate the integral of the Novosibirsk PDF.
+
+    Args:
+        peak: peak of the distribution.
+        width: width of the distribution.
+        tail: tail of the distribution.
+        lower: lower limit of the integral.
+        upper: upper limit of the integral.
+
+    Returns:
+        `tf.Tensor`: The calculated integral.
+
+    Notes:
+        Based on code from `ROOT <https://root.cern.ch/doc/master/Novosibirsk_8cxx_source.html>`_
```

Comment:
```suggestion
        Based on code from `ROOT <https://root.cern.ch/doc/master/RooNovosibirsk_8cxx_source.html>`_
```
Did you mean the RooNovosibirsk I assume?